src/classmentors/components/challenges/challenges.js

```diff
@@ -185,20 +183,80 @@ export function challengeServiceFactory
         delete copy.singPathProblem;
         delete copy.badge;
         delete copy.answers;
+        if(copy.link == ""){
+          delete copy.link;
+        }
         console.log(copy);
         // Create reccord in eventTeams
-
-        // Create reccord in answers and tasks
-        var ref = clmDataStore.events.addTaskWithAns(event.$id, copy, isOpen,answers);
-        ref.then(function() {
-            spfAlert.success('Task created');
-            $location.path(urlFor('editEvent', {eventId: event.$id}));
-        }).catch(function(err) {
-            $log.error(err);
-            spfAlert.error('Failed to created new task');
-        }).finally(function() {
-            self.creatingTask = false;
+        // Add IRAT
+        clmDataStore.events.addTaskWithAns(event.$id, copy, isOpen,answers)
+          .then(function(ref){
+            //add Team formation task
+            var previousTaskId = ref.key();
+            console.log('IRAT key is :', previousTaskId);
+            clmDataStore.events.addTeamFormation(event.$id, {
+                taskFrom: previousTaskId,
+                title: copy.title,
+                description: "Click Below To Join Team",
+                formationPattern: true,
+                closedAt : {'.sv': 'timestamp'},
+                showProgress: copy.showProgress,
+                archived: false,
+                teamFormationMethod: copy.teamFormationMethod
+            }, copy.priority).then(function(teamRef){
+              //create teams here
+              var taskId = teamRef.key()
+              console.log('taskId: ', taskId);
+              console.log(event.teams);
+              spfFirebase.set(['classMentors/eventTeams', event.$id],taskId)
```

Comment:
Here, you trying to set `classMentors/eventTeams/$eventId` to a literal value...